File: samples/python/trustregistry_demo.py
import asyncio

from trinsic.account_service import AccountService
from trinsic.proto.services.trustregistry.v1 import (
    RegistrationStatus,
    RegisterIssuerRequest,
    RegisterVerifierRequest,
    CheckIssuerStatusRequest,
    CheckVerifierStatusRequest,
)
from trinsic.trustregistry_service import TrustRegistryService
from trinsic.trinsic_util import trinsic_config


async def trustregistry_demo():
    # setup
    account_service = AccountService(server_config=trinsic_config())
    account = await account_service.sign_in()
    service = TrustRegistryService(server_config=trinsic_config(account))

    # data
    https_schema_org = "https://schema.org/Card"
    https_example_com = "https://example.com"
    did_example_test = "did:example:test"
    # register issuer
    await service.register_issuer(
        request=RegisterIssuerRequest(
            did_uri=did_example_test,
            governance_framework_uri=https_example_com,
            credential_type_uri=https_schema_org,
        )
    )

    # register verifier
    await service.register_verifier(
        request=RegisterVerifierRequest(
            did_uri=did_example_test,
            governance_framework_uri=https_example_com,
            presentation_type_uri=https_schema_org,
        )
    )

    # check issuer status
    check_response = await service.check_issuer_status(
        request=CheckIssuerStatusRequest(
            did_uri=did_example_test,
            governance_framework_uri=https_example_com,
            credential_type_uri=https_schema_org,
        )
    )
    issuer_status = check_response.status
    assert issuer_status == RegistrationStatus.CURRENT

    # check verifier status
    check_response = await service.check_verifier_status(
        request=CheckVerifierStatusRequest(
            did_uri=did_example_test,
            governance_framework_uri=https_example_com,
            presentation_type_uri=https_schema_org,
        )
    )
    verifier_status = check_response.status
    assert verifier_status == RegistrationStatus.CURRENT

    # search registry
    search_result = await service.search_registry()
    assert search_result is not None
    assert search_result.items_json is not None
    assert len(search_result.items_json) > 0


if __name__ == "__main__":
    asyncio.run(trustregistry_demo())

File: samples/ruby/trust_registry_demo.rb
# frozen_string_literal: true

require 'trinsic_services'
require 'securerandom'

# rubocop:disable Metrics/MethodLength
def trust_registry_demo_run
  trinsic = Trinsic::TrinsicService.new(Trinsic.trinsic_server)
  account = trinsic.account_service.login_anonymous

  trinsic.auth_token = account

  # New governance framework
  framework_uri = "urn:egf:#{SecureRandom.uuid}"
  governance_framework_response = trinsic.trust_registry_service.add_framework(Trinsic::TrustRegistry::AddFrameworkRequest.new(
                                                          governance_framework_uri: framework_uri, name: "Test Governance Framework - #{SecureRandom.uuid}"
                                                        ))

  # register issuer
  did_uri = 'did:example:test'
  type_uri = 'https://schema.org/Card'
  trinsic.trust_registry_service.register_member(Trinsic::TrustRegistry::RegisterMemberRequest.new(did_uri: did_uri,
                                                                               framework_id: governance_framework_response.id, schema_uri: type_uri))

  # check issuer status
  issuer_status = trinsic.trust_registry_service.get_membership_status(Trinsic::TrustRegistry::GetMembershipStatusRequest.new(
                                                  did_uri: did_uri, governance_framework_uri: framework_uri, schema_uri: type_uri
                                                ))
  raise "Issuer status #{issuer_status.status} should be current" unless issuer_status.status == :CURRENT

  # search registry
  search_result = trinsic.trust_registry_service.search
  raise 'Search result should exist' if search_result.nil?
  raise 'Search result should not be empty' unless JSON.parse(search_result.items_json).length.positive?
end
# rubocop:enable Metrics/MethodLength

trust_registry_demo_run
